### ISSUE TITLE
Commands for managing groups with occ.

### ISSUE THREAD

**Issue by SergioBertolinSG** [owncloud/core#20629]
Currently there are no options for managing groups:

```
ownCloud version 9.0 pre alpha

Usage:
 [options] command [arguments]

Options:
 --help (-h)           Display this help message
 --quiet (-q)          Do not output any message
 --verbose (-v|vv|vvv) Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug
 --version (-V)        Display this application version
 --ansi                Force ANSI output
 --no-ansi             Disable ANSI output
 --no-interaction (-n) Do not ask any interactive question

Available commands:
 check                                check dependencies of the server environment
 help                                 Displays help for a command
 list                                 Lists commands
 status                               show some status information
 upgrade                              run upgrade routines after installation of a new release. The release has to be installed before.
app
 app:check-code                       check code to be compliant
 app:disable                          disable an app
 app:enable                           enable an app
 app:list                             List all available apps
background
 background:ajax                      Use ajax to run background jobs
 background:cron                      Use cron to run background jobs
 background:webcron                   Use webcron to run background jobs
config
 config:app:delete                    Delete an app config value
 config:app:get                       Get an app config value
 config:app:set                       Set an app config value
 config:import                        Import a list of configs
 config:list                          List all configs
 config:system:delete                 Delete a system config value
 config:system:get                    Get a system config value
 config:system:set                    Set a system config value
dav
 dav:create-addressbook               Create a dav addressbook
 dav:create-calendar                  Create a dav calendar
db
 db:convert-type                      Convert the ownCloud database to the newly configured one
 db:generate-change-script            generates the change script from the current connected db to db_structure.xml
encryption
 encryption:change-key-storage-root   Change key storage root
 encryption:decrypt-all               Disable server-side encryption and decrypt all files
 encryption:disable                   Disable encryption
 encryption:enable                    Enable encryption
 encryption:encrypt-all               Encrypt all files for all users
 encryption:list-modules              List all available encryption modules
 encryption:set-default-module        Set the encryption default module
 encryption:show-key-storage-root     Show current key storage root
 encryption:status                    Lists the current status of encryption
files
 files:cleanup                        cleanup filecache
 files:scan                           rescan filesystem
l10n
 l10n:createjs                        Create javascript translation files for a given app
log
 log:manage                           manage logging configuration
 log:owncloud                         manipulate ownCloud logging backend
maintenance
 maintenance:mimetype:update-db       Update database mimetypes and update filecache
 maintenance:mimetype:update-js       Update mimetypelist.js
 maintenance:mode                     set maintenance mode
 maintenance:repair                   repair this installation
 maintenance:singleuser               set single user mode
trashbin
 trashbin:cleanup                     Remove deleted files
user
 user:add                             adds a user
 user:delete                          deletes the specified user
 user:lastseen                        shows when the user was logged it last time
 user:report                          shows how many users have access
 user:resetpassword                   Resets the password of the named user
versions
 versions:cleanup                     Delete versions
```

This would be very nice to have.

**Comment by PVince81** [comment]
Sometimes I wonder if we should write a separate tool like OCC but that connects to the provisioning API (or other APIs) to perform operations. That would remove the need to have duplicate APIs in OCC and others and would have the benefit of working remotely (like pyocclient)

**Comment by edwardcrompton** [comment]
This seems a little bit more involved than some of the other junior jobs I've had a look at. However, I thought I'd have a go at creating an add group command. I can expand this to other commands but it would be good to have a bit of reassurance I'm approaching it in the right way first. Would you mind taking a look at this pull request https://github.com/owncloud/core/pull/21627/files @PVince81 @DeepDiver1975 ?

**Comment by phil-davis** [comment]
Resolved by the above PRs #28190 and #28318 
Docs are done:
 https://github.com/owncloud/documentation/pull/3201
https://github.com/owncloud/documentation/pull/3208

@SergioBertolinSG OK to close?
